src/sentry/rules/filters/age_comparison.py

```diff
@@ -73,11 +72,37 @@ def passes(self, event: GroupEvent, state: EventState) -> bool:
             )
         ):
             return False
+        return value
 
+    def _compare(self, first_seen, comparison_type, time, value):
         _, delta_time = timeranges[time]
-
-        first_seen = event.group.first_seen
-        passes_: bool = age_comparison_map[comparison_type](
+        return age_comparison_map[comparison_type](
             first_seen + (value * delta_time), timezone.now()
         )
-        return passes_
+
+    def passes(self, event: GroupEvent, state: EventState) -> bool:
+        comparison_type = self.get_option("comparison_type")
+        time = self.get_option("time")
+        value = self.get_option("value")
+
+        value = self._validate(comparison_type, time, value)
+        if value is False:
+            return False
+
+        return self._compare(event.group.first_seen, comparison_type, time, value)
+
+    def passes_activity(self, condition_activity):
+        comparison_type = self.get_option("comparison_type")
+        time = self.get_option("time")
+        value = self.get_option("value")
+
+        value = self._validate(comparison_type, time, value)
+        if value is False:
+            return False
+
+        try:
+            group = Group.objects.get_from_cache(id=condition_activity.group_id)
+        except Group.DoesNotExist:
+            return False
+
+        return self._compare(group.first_seen, comparison_type, time, value)
```

Comment:
maybe add `condition_activity` as a named arg with default `None`. if the variable is present add the group check.

these functions are basically doing the same thing

Reply:
and then like set a `first_seen` variable that is conditional based on that logic